src/settings.h

```diff
@@ -257,7 +286,8 @@ class Settings {
 	// All methods that access m_settings/m_defaults directly should lock this.
 	mutable std::mutex m_mutex;
 
-	static Settings *s_layers[SL_TOTAL_COUNT];
-	SettingsLayer m_settingslayer = SL_TOTAL_COUNT;
+	SettingsHierarchy *m_hierarchy = nullptr;
+	int m_settingslayer = -1;
```

Comment:
u32

Reply:
How would I set this to -1 if it's unsigned? (That is supposed to represent an invalid layer number)

Reply:
@sfan5 is this normal to have a Setting without layer ? i think we can have one layer by default, else this has no sense. Do you have usecase without layers ?

Reply:
For me layer 0 is the root.

```
setting name: setting.aaa.bbb.ccc
setting layer:      0   1   2   3
```

Reply:
Setting objects can exist without being bound to any hierarchy, then it makes no sense to define any layer.
I don't know what the second post is supposed to mean.